app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/state/MessageCompositionInputStateHolder.kt

```diff
@@ -19,28 +19,55 @@ package com.wire.android.ui.home.messagecomposer.state
 
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.Stable
 import androidx.compose.runtime.State
 import androidx.compose.runtime.derivedStateOf
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.saveable.Saver
 import androidx.compose.runtime.setValue
+import androidx.compose.ui.focus.FocusManager
 import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.max
+import com.google.android.gms.common.util.VisibleForTesting
 import com.wire.android.R
 import com.wire.android.ui.common.colorsScheme
 import com.wire.android.ui.common.textfield.WireTextFieldColors
 import com.wire.android.ui.common.textfield.wireTextFieldColors
+import com.wire.android.util.ui.KeyboardHeight
 import com.wire.kalium.logic.feature.selfDeletingMessages.SelfDeletionTimer
 import com.wire.kalium.logic.util.isPositiveNotNull
 
+@Stable
 class MessageCompositionInputStateHolder(
     private val messageComposition: MutableState<MessageComposition>,
     selfDeletionTimer: State<SelfDeletionTimer>
 ) {
     var inputFocused: Boolean by mutableStateOf(false)
         private set
 
+    var keyboardHeight by mutableStateOf(KeyboardHeight.default)
+        private set
+
+    var optionsHeight by mutableStateOf(0.dp)
+        private set
+
+    var showOptions by mutableStateOf(false)
```

Comment:
suggestion: it would be easier to understand if it would be an adjective instead of noun, because it's not an action but a state, for example: "optionsShown" or "optionsVisible"

Reply:
Good idea